lightstep/resource_metric_dashboard.go

```diff
@@ -207,15 +238,44 @@ func getUnifiedDashboardAttributesFromResource(d *schema.ResourceData) (*client.
 		return nil, err
 	}
 
+	templateVariableSet := d.Get("template_variable").(*schema.Set)
+	templateVariables := buildTemplateVariables(templateVariableSet.List())
+
 	attributes := &client.UnifiedDashboardAttributes{
-		Name:        d.Get("dashboard_name").(string),
-		Description: d.Get("dashboard_description").(string),
-		Charts:      charts,
+		Name:              d.Get("dashboard_name").(string),
+		Description:       d.Get("dashboard_description").(string),
+		Charts:            charts,
+		TemplateVariables: templateVariables,
 	}
 
 	return attributes, nil
 }
 
+func buildTemplateVariables(templateVariablesIn []interface{}) []client.TemplateVariable {
+	var newTemplateVariables []client.TemplateVariable
+	for _, tv := range templateVariablesIn {
+		tvMap := tv.(map[string]interface{})
+		name := tvMap["name"].(string)
+		suggestionAttributeKey := tvMap["suggestion_attribute_key"].(string)
+		defaultValues := buildDefaultValues(tvMap["default_values"].([]interface{}))
+
+		newTemplateVariables = append(newTemplateVariables, client.TemplateVariable{
+			Name:                   name,
+			DefaultValues:          defaultValues,
+			SuggestionAttributeKey: suggestionAttributeKey,
+		})
+	}
+	return newTemplateVariables
+}
+
+func buildDefaultValues(valuesIn []interface{}) []string {
+	var defaultValues []string
+	for _, v := range valuesIn {
+		defaultValues = append(defaultValues, v.(string))
```

Comment:
Is the user type, say `4` rather than `"4"` in the Terraform file as the value for the template variable, what happens? Will it end up with a panic here? 

I know the existing code doesn't set a great precedent for good user error messaging, but is there some way we can give a more informative error (assuming it does panic if a number were provided instead of a string)?

Reply:
From what I remember testing earlier this week, this gets converted to `"4"` by the provider, but worth to double check :)